### PR TITLE
Feature/python3 11 support issue #185

### PR DESCRIPTION
**Pull request by carlkidcrypto**
@kamakazikamikaze  This PR addresses issue #185. I bumped the version number to 0.3.0a since it will only support python3 versions >= 3.8.

The merge request is into the branch "test". Not sure if you want to create a new branch off of "test" and call it something like "release_v0.3.0_integration". I'll leave that up to you.

All tests pass as well as Valgrind. (Ran the workflows on my fork branch).

I also added a dependabot workflow to keep "github-actions" stuff up to date.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 name: tests
 on:
   push:
-    branches-ignore:
-      - 'master'
-      - 'actions'
+    branches: [ feature/python3_11_support_issue_#185 ]
```
Comment (carlkidcrypto) [inline .github/workflows/tests.yml]: Undo this change. no longer needed.

Comment (carlkidcrypto) [reply .github/workflows/tests.yml]: Done in - [80883e9](https://github.com/easysnmp/easysnmp/pull/186/commits/80883e9252851b14f8fbf6b1a3bfc9d342049212)

**Comment by mjbogusz** [comment]
Bumping for visibility, I'm using easysnmp and I'd love to be able to migrate my project to newer python versions.
Is there something I could do to help merging this?

Additionally, would it be possible to add 3.12 ([dropped 2 days ago](https://www.python.org/downloads/release/python-3120/)) here as well for a bit of future-proofing?

**Comment by carlkidcrypto** [comment]
@mjbogusz I just pushed support for Python 3.12. I haven't actually tested it on 3.12. I'll have to spin up my setup again at some point with 3.12. But if you already have one up and running go ahead and give this branch a try.

You should be able to switch to this branch and install it from source.

**Comment by carlkidcrypto** [comment]
@mjbogusz Looks like `distutils` was removed in Python 3.12 So this code won't install on Python 3.12. 

Note to self: Follow this page to move away from `distutils` https://peps.python.org/pep-0632/#migration-advice

**Comment by h5rdly** [comment]
Hi,

Sorry for the bump, didn't want to open a new issue for this.

I was able to `pip3 install easysnmp` on kubuntu 22.04 with python 3.11, after running `sudo apt install -y libsnmp-dev python3.11-dev`.

However, the repo doesn't indicate it, e.g https://github.com/easysnmp/easysnmp/blob/test/setup.py only mentions Python 3.10 as supported.

Would it be possible to update `setup.py`, and/or add those little green labels at the beginning of the readme with supported python versions? 

Might help newcomers like myself :)


Thanks,
Eli

**Comment by mjbogusz** [comment]
I'm afraid that adding a list of supported python versions to readme is possible only for the maintainer, @kamakazikamikaze, but as there were no updates to this repo in 2 years, I wouldn't hold my breath.

I've switched my project to https://github.com/carlkidcrypto/ezsnmp (available in pypi as well), I'm yet to run a full regression/performance suite but so far everything looks a-ok.

**Comment by h5rdly** [comment]
Thanks, was not aware of this repo!

Would love to hear if you found any regressions, seems the code was moved from C to C++ 17.

**Comment by mjbogusz** [comment]
Unfortunately my project that was using easysnmp has been pushed to backlog, so I won't be able to provide a better answer for some weeks.

However, after merely a quick search&replace of `easysnmp`->`ezsnmp` (requirements, imports, exceptions - mind the letter casing!) all my unit tests were green.
I'm using only `snmp_get` and `Session.walk()` though, so there may be incompatibilities elsewhere.

**Comment by mjbogusz** [comment]
Unexpectedly I was able to work on this sooner than I thought and the results are in.

Note that my usage of the library is limited to methods mentioned in previous comment and that I'm not following the best practice of opening and keeping one `Session` object - in my typical usecase it's not necessary and for this benchmarking it served as a torture test, but it might have influenced the outcome.

My benchmark consisted of measuring the time it took to query a real physical device (network switch) for a single and multiple specific OIDs 1000 times in a row with `snmp_get()`, and querying 2 OID subtrees with about 30 objects each  50 times with `Session.walk()`.

I've ran all 3 variants 10 times (with some pause in between runs) for both libraries and after dismissing 2 worst results for each combination to account for network congestion and such, and averaging the results, here's the difference:
* single OID `snmp_get()`: ezsnmp -12.3% (as in: 12.3% better time on average)
* multiple OIDs `snmp_get()`: ezsnmp -9.1%
* 2 subtrees x 30 objects `Session.walk()`: ezsnmp -0.6%

All in all I'm very happy with the results, and while I will NOT say that ezsnmp is definitely faster because my method might be flawed, at least it should not be significantly slower (like PySNMP) and so for my project I'll be fully switching to ezsnmp.
